summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SummerBoltProvider.scala

```diff
@@ -12,14 +15,34 @@ import com.twitter.summingbird.storm.planner.StormNode
 import org.slf4j.LoggerFactory
 import scala.reflect.ClassTag
 
-case object SummerBoltProvider {
+private[storm] case object SummerBoltProvider {
   @transient private val logger = LoggerFactory.getLogger(SummerBoltProvider.getClass)
 }
 
-case class SummerBoltProvider(builder: StormTopologyBuilder, node: SummerNode[Storm]) {
+private[storm] case class SummerBoltProvider(
+  builder: StormTopologyBuilder,
+  node: SummerNode[Storm]
+) extends ComponentProvider {
   import SummerBoltProvider._
 
-  def apply[K, V]: SummerBolt[K, V] = {
+  override def createSingle[T, O](fn: Item[T] => O): Topology.Component[O] = {
+    def create[K, V](fn: Item[(K, (Option[V], V))] => O): Topology.Component[O] =
+      bolt[K, V, O](fn)
+    // This is a legitimate conversion because we now that summer emits (K, (Option[V], V)).
```

Comment:
I had a version before where I had two separate methods `createSingle` (for `Producer[T]` basically) and `createKeyValue` for (`KeyedProducer[K, V]`). And Summer wasn't implementing first one for example. But I removed it because of complexity. I think this way it's just simpler and there are not a lot of unsafe casts in code as well so this one is ok.